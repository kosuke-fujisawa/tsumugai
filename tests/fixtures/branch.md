[SAY speaker=Guide]
Choose your path.

[BRANCH choice=Left label=go_left, choice=Right label=go_right]

[LABEL name=go_left]

[SAY speaker=Guide]
You went left.

[LABEL name=go_right]

[SAY speaker=Guide]
You went right.