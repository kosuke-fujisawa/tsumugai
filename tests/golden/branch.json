{
  "input_hash": "4775c9f8047b4b20c8fbe30e04f6c1a7",
  "steps": [
    {
      "step_number": 0,
      "next_action": "WaitUser",
      "directives": [
        {
          "type": "Say",
          "args": {
            "speaker": "Guide",
            "text": "Choose your path."
          }
        }
      ]
    },
    {
      "step_number": 1,
      "next_action": "WaitBranch",
      "directives": [
        {
          "type": "Branch",
          "args": {
            "choices": [
              "Left",
              "Right"
            ]
          }
        }
      ]
    },
    {
      "step_number": 2,
      "next_action": "Next",
      "directives": [
        {
          "type": "ReachedLabel",
          "args": {
            "label": "go_left"
          }
        }
      ]
    },
    {
      "step_number": 3,
      "next_action": "WaitUser",
      "directives": [
        {
          "type": "Say",
          "args": {
            "speaker": "Guide",
            "text": "You went left."
          }
        }
      ]
    },
    {
      "step_number": 4,
      "next_action": "Next",
      "directives": [
        {
          "type": "ReachedLabel",
          "args": {
            "label": "go_right"
          }
        }
      ]
    },
    {
      "step_number": 5,
      "next_action": "WaitUser",
      "directives": [
        {
          "type": "Say",
          "args": {
            "speaker": "Guide",
            "text": "You went right."
          }
        }
      ]
    },
    {
      "step_number": 6,
      "next_action": "Halt",
      "directives": []
    }
  ],
  "final_state": "Completed"
}
